pkg/tms/tmsUtils.go

```diff
@@ -140,6 +146,14 @@ func unmarshalServiceKey(serviceKeyJson string) (serviceKey serviceKey, err erro
 	if err != nil {
 		return
 	}
+	if len(serviceKey.Uri) == 0 {
+		if serviceKey.CALMEndpoints != nil && len(serviceKey.CALMEndpoints.API) > 0 {
```

Comment:
Do you need the check for nil? Wouldn't len(nil) return 0
```suggestion
		if len(serviceKey.CALMEndpoints.API) > 0 {
```

Reply:
We have to check that the `CALMEndpoints` exists before we can look inside of it. With the suggested code the following error is thrown, because it tried to look at `API`, which is inside `CALMEndpoints`, which is `nil`:
> panic: runtime error: invalid memory address or nil pointer dereference

I don't know if there is a more elegant way of doing this.